### PR TITLE
#1031 Set up basic docusaurus structure

### PR DESCRIPTION
**Pull request by sveinpg**
#1031.  Initial setup of docusaurus in a separate folder (`/docusaurus`). 

To run locally or publish you first have to change folders to `/docusaurus/website`

Run locally: `npm install && npm start`
This should open the docusaurus on localhost:3000

To publish to netlify: https://docusaurus.io/docs/en/publishing#hosting-on-netlify

### CODE REVIEW

**Comment by sveinpg** [comment]
@markerikson Seems like some checks were not successful? Do you mind pointing me in the right direction?

File: docusaurus/website/siteConfig.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  tagline: "React Redux Documentation",
+  url: "https://your-docusaurus-test-site.com", // Your website URL
+  baseUrl: "/", // Base URL for your project */
+  // For github.io type URLs, you would set the url and baseUrl like:
+  //   url: 'https://facebook.github.io',
+  //   baseUrl: '/test-site/',
+
+  // Used for publishing and more
+  projectName: "react-redux",
+  organizationName: "facebook",
```
Comment (markerikson) [inline docusaurus/website/siteConfig.js]: We're not Facebook :)  We're the `reduxjs` org.

File: docusaurus/website/siteConfig.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// See https://docusaurus.io/docs/site-config for all the possible
+// site configuration options.
+
+const siteConfig = {
+  title: "React Redux", // Title for your website.
+  tagline: "React Redux Documentation",
+  url: "https://your-docusaurus-test-site.com", // Your website URL
```
Comment (markerikson) [inline docusaurus/website/siteConfig.js]: This should hopefully be `https://react-redux.js.org` shortly.

File: docusaurus/website/siteConfig.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ],
+
+  /* path to images for header/footer */
+  // headerIcon: "img/redux.svg",
+  // footerIcon: "img/redux.svg",
+  favicon: "img/favicon/favicon.ico",
+
+  /* Colors for website */
+  colors: {
+    primaryColor: "#20232a",
```
Comment (markerikson) [inline docusaurus/website/siteConfig.js]: Seems like these are shades of blue?  Let's try to go with Redux's purple and white, or something like that.

**Comment by markerikson** [comment]
Now that I look at this, is there a reason to actually have this in a `/docusaurus` subfolder?  Babel's setup doesn't do that - it's just `$REPO/website` ( https://github.com/babel/website ).

**Comment by sveinpg** [comment]
Babel's setup is a new repo for the docusaurus docs. 

Personally I prefer if the docs is in the repo itself. However, babel is a special case since it consists of multiple repos. 

We could move the `website` folder to the root of this repo?

**Comment by sveinpg** [comment]
The `dockerfile` was generated when I ran the setup, we don't need it.

**Comment by markerikson** [comment]
Okay, let's give this a shot.  Can always revert it if things seem weird.

I'll merge it now.  I'll be busy this afternoon, but will try working on some of the setup for it tonight.

**Comment by markerikson** [comment]
Okay, this is pretty cool!  I just pulled master locally, installed website deps, and ran the local server, and it works!  

We'll want to see how to get better syntax highlighting for the snippets, and also play around with how to get CodeSandbox embeds working, but this is a great start!

**Comment by sveinpg** [comment]
Great! Let me know if I can contribute on anything else.

**Comment by markerikson** [comment]
Tell you what.  Can you go and investigate how Babel and React have a bot set up to do preview deploys of PRs?

**Comment by markerikson** [comment]
WOOHOO!  We are **LIVE!!!**

https://react-redux-docs.netlify.com/

**Comment by sveinpg** [comment]
Sure. I'll take a look at it tomorrow

**Comment by sveinpg** [comment]
@markerikson Seems like you can enable branch deploys to get the desired behaviour as described here:  https://www.netlify.com/docs/continuous-deployment/#branches-deploys

**Comment by sveinpg** [comment]
There is a section om how to configure it called "github pull request comments"

https://www.netlify.com/docs/webhooks/#outgoing-webhooks

**Comment by markerikson** [comment]
Yep, got that figured out and turned on!

**Pull request merged**